qiskit/primitives/containers/observable.py

```diff
@@ -0,0 +1,188 @@
+            return cls.coerce(data)
+
+        if isinstance(observable, Pauli):
+            label, phase = observable[:].to_label(), observable.phase
+            cls._validate_basis(label)
+            data = {label: 1} if phase == 0 else {label: (-1j) ** phase}
+            return Observable(data)
+
+        # String conversion
+        if isinstance(observable, str):
+            cls._validate_basis(observable)
+            return cls.coerce({observable: 1})
+
+        # Mapping conversion (with possible Pauli keys)
+        if isinstance(observable, Mapping):
+            # NOTE: This assumes length of keys is number of qubits
+            #       this might not be very robust
+            num_qubits = len(next(iter(observable)))
+            unique = defaultdict(complex)
+            for basis, coeff in observable.items():
+                if isinstance(basis, Pauli):
+                    basis, phase = basis[:].to_label(), basis.phase
+                    if phase != 0:
+                        coeff = coeff * (-1j) ** phase
+                # Validate basis
+                cls._validate_basis(basis)
+                if len(basis) != num_qubits:
+                    raise ValueError(
+                        "Number of qubits must be the same for all observable basis elements."
+                    )
+                unique[basis] += coeff
+            return Observable(dict(unique))
+
+        raise TypeError(f"Invalid observable type: {type(observable)}")
+
+    @classmethod
+    def _validate_basis(cls, basis: any) -> None:
+        """Validate a basis string.
+
+        Args:
+            basis: a basis object to validate.
+
+        Raises:
+            TypeError: If the input basis is not a string
+            ValueError: If basis string contains invalid characters
+        """
+        if not isinstance(basis, str):
+            raise TypeError(f"basis {basis} is not a string")
+
+        # NOTE: the allowed basis characters can be overridden by modifying the class
+        # attribute ALLOWED_BASIS
+        allowed_pattern = _regex_match(cls.ALLOWED_BASIS)
+        if not allowed_pattern.match(basis):
+            invalid_pattern = _regex_invalid(cls.ALLOWED_BASIS)
+            invalid_chars = list(set(invalid_pattern.findall(basis)))
+            raise ValueError(
+                f"Observable basis string '{basis}' contains invalid characters {invalid_chars},"
+                f" allowed characters are {list(cls.ALLOWED_BASIS)}.",
+            )
+
+    @classmethod
+    def _validate_coeff(cls, coeff: any):
+        """Validate the consistency in observables array."""
+        if not isinstance(coeff, Complex):
+            raise TypeError(f"Value {coeff} is not a complex number")
+
+
+@lru_cache(1)
+def _regex_match(allowed_chars: str) -> re.Pattern:
+    """Return pattern for matching if a string contains only the allowed characters."""
+    return re.compile(f"^[{re.escape(allowed_chars)}]*$")
```

Comment:
I'm wondering whether an empty string is allowed, e.g.,
```python
print(Observable({'': 1}))
# Observable({'': 1})
```

If no, I suggest updating as follows.
```suggestion
    return re.compile(f"^[{re.escape(allowed_chars)}]+$")
```